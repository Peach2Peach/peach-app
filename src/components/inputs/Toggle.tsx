import { TextStyle, TouchableOpacity } from 'react-native'
import tw from '../../styles/tailwind'
import { Icon } from '../Icon'
import Text from '../text/Text'

type Props = ComponentProps & {
  enabled: boolean
  disabled?: boolean
  onPress: () => void
  textStyle?: TextStyle
}
export const Toggle = ({ enabled, disabled, onPress, children, style, textStyle }: Props) => (
  <TouchableOpacity
    style={[tw`flex-row items-center gap-4`, disabled && tw`opacity-33`, style]}
    onPress={onPress}
    disabled={disabled}
  >
    {!!children && <Text style={[tw`settings`, textStyle]}>{children}</Text>}
    {enabled ? (
      <Icon id="toggleRight" size={32} color={tw.color('primary-main')} />
    ) : (
      <Icon id="toggleLeft" size={32} color={tw.color('black-3')} />
    )}
  </TouchableOpacity>
)
