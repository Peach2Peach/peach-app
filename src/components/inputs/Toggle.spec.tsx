import { fireEvent, render } from '@testing-library/react-native'
import { createRenderer } from 'react-test-renderer/shallow'
import tw from '../../styles/tailwind'
import { Toggle } from './Toggle'

describe('Toggle', () => {
  const renderer = createRenderer()
  it('renders correctly when not enabled', () => {
    renderer.render(
      <Toggle enabled={false} onPress={jest.fn()}>
        label
      </Toggle>,
    )
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
  it('renders correctly when enabled', () => {
    renderer.render(
      <Toggle enabled={true} onPress={jest.fn()} style={tw`mt-4`}>
        label
      </Toggle>,
    )
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })

  it('calls onPress handler', () => {
    const onPress = jest.fn()
    const { getByText } = render(
      <Toggle enabled={true} onPress={onPress}>
        label
      </Toggle>,
    )
    fireEvent.press(getByText('label'))
    expect(onPress).toHaveBeenCalled()
  })
})
