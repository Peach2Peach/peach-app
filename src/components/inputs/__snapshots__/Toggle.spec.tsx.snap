// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle renders correctly when enabled 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  delayLongPress={600}
  extraButtonProps={
    {
      "exclusive": true,
      "rippleColor": "transparent",
    }
  }
  onPress={[MockFunction]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 16,
      },
      {
        "marginTop": 16,
      },
    ]
  }
>
  <PeachText
    style={
      {
        "fontFamily": "Baloo2-SemiBold",
        "fontSize": 20,
        "letterSpacing": 0.2,
        "lineHeight": 32,
        "textTransform": "lowercase",
      }
    }
  >
    label
  </PeachText>
  <Icon
    color="#F56522"
    id="toggleRight"
    style={
      {
        "height": 32,
        "width": 32,
      }
    }
  />
</TouchableOpacity>
`;

exports[`Toggle renders correctly when not enabled 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  delayLongPress={600}
  extraButtonProps={
    {
      "exclusive": true,
      "rippleColor": "transparent",
    }
  }
  onPress={[MockFunction]}
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 16,
      },
      undefined,
    ]
  }
>
  <PeachText
    style={
      {
        "fontFamily": "Baloo2-SemiBold",
        "fontSize": 20,
        "letterSpacing": 0.2,
        "lineHeight": 32,
        "textTransform": "lowercase",
      }
    }
  >
    label
  </PeachText>
  <Icon
    color="#9F8C82"
    id="toggleLeft"
    style={
      {
        "height": 32,
        "width": 32,
      }
    }
  />
</TouchableOpacity>
`;
